tool/tsh/tsh.go

```diff
@@ -1005,16 +1014,19 @@ func loadIdentity(idFn string) (*client.Key, ssh.HostKeyCallback, error) {
 	}
 	defer f.Close()
 	var (
-		keyBuf  bytes.Buffer
-		state   int // 0: not found, 1: found beginning, 2: found ending
-		cert    []byte
-		caCerts [][]byte
+		keyBuf     bytes.Buffer
+		state      int // 0: init, 1: reading key, 2: not reading, 3: reading tls cert
+		cert       []byte
+		tlsCertBuf bytes.Buffer
+		tlsCert    []byte
+		tlsCACerts [][]byte
+		caCerts    [][]byte
 	)
 	// read the identity file line by line:
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
 		line := strings.TrimSpace(scanner.Text())
-		if state != 1 {
+		if state%2 != 1 { // Odd numbered state means that we are in the middle of a read
```

Comment:
let's just switch to proper constants instead of magic numbers/properties

Reply:
Are both the key and TLS certificate PEM encoded? If they are, you can just use the PEM decoder from the standard library and let decode them for you without doing any parsing here. Take a look:  https://golang.org/pkg/encoding/pem/#Decode

If they are not, I second the suggestion to use constants, also, can you use a switch statement here, will make it easier to follow:
```go
switch state {
case ReadingSSH, ReadingTLS:
   ...
case NotReading:
  ...
}
```

Reply:
Ended up just eliminating the need for state variables entirely by consuming all lines of the PEM block within a single iteration of the loop.